static/app/components/dynamicSampling/investigationRule.tsx

```diff
@@ -174,34 +178,20 @@ const InvestigationInProgressNotification = styled('span')`
   gap: ${space(0.5)};
 `;
 
-function checkIsTransactionQueryMissing(error: RequestError | null) {
-  if (error?.responseJSON?.query) {
-    const query = error.responseJSON.query;
-    if (Array.isArray(query)) {
-      for (const reason of query) {
-        if (reason === 'not_transaction_query') {
-          return true;
-        }
-      }
-    }
-  }
-  return false;
-}
-
 function InvestigationRuleCreationInternal(props: PropsInternal) {
   const projects = [...props.eventView.project];
   const organization = props.organization;
   const period = props.eventView.statsPeriod || null;
   const query = props.eventView.getQuery();
+  const isTransactionQueryMissing =
+    getQueryDatasource(query)?.source !== Datasource.TRANSACTION;
   const createInvestigationRule = useCreateInvestigationRuleMutation();
   const {
     data: rule,
-    isLoading,
+    isFetching: isLoading,
```

Comment:
had to use `isFetching` instead of `isLoading`, because, in this current version of react-query, the `isLoading` state is always `true` if `enabled` equals `false`